### DISCUSSION TITLE
Nested targets with similar names are incorrectly triggered when loading

### DISCUSSION THREAD

**Discussion by solarsilk** [livewire/livewire#7485] [7. Bugs (archived)]
### Livewire version

3

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

When I target 2 similarly-named nested properties, and one is in the loading state, the other is also put in the loading state.

Eg. Below, I'm using an array property (`content`) with 2 nested properties with similar names. When `content.test_2` is loading, `content.test` is also put in the loading state.

```
<input type="text"	
wire:loading.attr="disabled"
wire:target="content.test"
wire:model.blur="content.test"
>

<input type="text"	
wire:loading.attr="disabled"
wire:target="content.test_2"
wire:model.blur="content.test_2"
>
```

I believe this is caused by [this line](https://github.com/livewire/livewire/blob/ecded08cdc4b36bbb4b26bcc7f7a171ea2e4368c/js/directives/wire-loading.js#L121C47-L121C47) which is
`property.startsWith(target)`

Should it be this instead?
`property.matches(target)`

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment]
@solarsilk thanks for reporting! If you have a potential fix, then PRs are welcome! Or even if you could submit a failing test PR, that would be great. Thanks!

**Reply by solarsilk** [reply]
@joshhanley  thanks for your reply. I haven't ever done a PR but I will give it a try!

**Reply by solarsilk** [reply]
@joshhanley  here's the PR:  https://github.com/livewire/livewire/pull/7531  Not sure if I did it correctly, so please let me know if I need to change anything. Thanks!

**Reply by joshhanley** [reply]
@solarsilk great work, thanks! Have commented on the PR 🙂

**Comment by solarsilk** [comment]
update PR #7548
